Dart/src/com/jetbrains/lang/dart/util/DartResolveUtil.java

```diff
@@ -1030,8 +1030,13 @@ public static DartClass suggestType(PsiElement element) {
 
   @Nullable
   public static DartComponent findReferenceAndComponentTarget(@Nullable PsiElement element) {
-    DartReference reference = PsiTreeUtil.getParentOfType(element, DartReference.class);
-    PsiElement target = reference == null ? null : reference.resolve();
+    PsiElement target;
+    if (element instanceof DartReferenceExpression) {
+      target = ((DartReferenceExpression)element).resolve();
```

Comment:
`DartReferenceExpression` extends `DartReference`, I think instead of a separate if-branch it's better to change `getParentOfType` to `getNonStrictParentOfType`